ichnaea/locate/provider.py

```diff
@@ -611,16 +614,34 @@ def limit_reached(self):
             self.redis_client,
             self.get_ratelimit_key(),
             maxreq=self.ratelimit,
-            expire=60,
+            expire=self.RATE_LIMIT_EXPIRE,
             fail_on_error=True,
         )
 
-    def locate(self, data):
-        location = self.location_type(query_data=False)
+    def _get_cache_key(self, cell_data):
+        return 'fallback_cache_{}'.format(sorted(cell_data.items()))
 
-        if self.limit_reached():
-            return location
+    def _fetch_cached_result(self, data):
+        all_cell_data = data.get('cell', [])
 
+        if len(all_cell_data) == 1:
+            cache_key = self._get_cache_key(all_cell_data[0])
+            cached_cell = self.redis_client.get(cache_key)
+            if cached_cell:
+                return json.loads(cached_cell)
```

Comment:
Can you add one more counter to this, something like:
`self.stat_count('fallback.cache.hit')` and `self.stat_count('fallback.cache.miss')`